src/opustrainer/trainer.py

```diff
@@ -548,14 +548,14 @@ def _match_modifier_signature(self, fn:Type[Modifier], settings:Dict[str, Any],
 
     T = TypeVar('T')
 
-    def _dynamic_cast_parameter(self, typedef:Type[T], value:Any, basepath:str) -> T:
+    def _dynamic_cast_parameter(self, typedef:Type[T], value:Any, basepath:str) -> Union[T, None, Path]:
         """Casts `value` into a type that matches `typedef`. Can deal with `Union`."""
         if get_origin(typedef) == Union:
             attempt_errors = ''
             for subtype in get_args(typedef):
                 try:
                     return self._dynamic_cast_parameter(subtype, value, basepath)
-                except err:
+                except Exception as err:
```

Comment:
I think this was invalid syntax, unless you meant to include an err class?

Reply:
Yep. I just can't remember how exceptions work across Javascript/Python/C++ it seems :(